From bcbe82e981d23848c78809317c147d1b2e3bc709 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@suse.com>
Date: Sun, 11 May 2014 20:32:02 -0400
Subject: [PATCH] crowbar_register: Do not keep sledgehammer-specific
 configuration

When we first run chef-client, we need the FQDN to be resolvable, which
means we need it in /etc/hosts. But that's not something we want to
keep.

We also do not want to have "hostname" return the FQDN name anymore
after we're setup.
---
 .../cookbooks/provisioner/templates/suse/crowbar_register.erb | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb b/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb
index d59c3d1..c0f4c8c 100644
--- a/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb
+++ b/chef/cookbooks/provisioner/templates/suse/crowbar_register.erb
@@ -287,6 +287,17 @@ post_state $HOSTNAME "installing"
 rm -f /etc/chef/client.pem
 
 
+# Revert bits from sledgehammer that should not persist
+# -----------------------------------------------------
+
+# Now setup the hostname with the short name; we couldn't do that earlier
+# because we do like in sledgehammer (which sets the hostname to the FQDN one)
+hostname "${HOSTNAME%%.*}"
+
+# Remove the resolution for hostname in /etc/hosts; this shouldn't be needed anymore
+sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 localhost.localdomain localhost/" /etc/hosts
+
+
 # Steps from autoyast profile
 # ---------------------------
 echo "$CROWBAR_KEY" > /etc/crowbar.install.key
-- 
1.8.4.5

